File: Android_Java/N8-221402/ScrollView/app/src/main/java/com/example/scrollview/MainActivity.java
package com.example.scrollview;

import android.os.Bundle;
import android.view.View;
import android.widget.ScrollView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        AppCompatButton btn_ScrollUp = findViewById(R.id.btn_ScrollUp);
        AppCompatButton btn_scrollDown = findViewById(R.id.btn_ScrollDown);
        ScrollView scrollView = findViewById(R.id.scrollView);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        btn_ScrollUp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int targetY = 0;
                scrollView.fullScroll(scrollView.FOCUS_UP);
            }
        });
        btn_scrollDown.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                scrollView.fullScroll(scrollView.FOCUS_DOWN);
            }
        });
    }
}

File: android-kotlin/RoomDatabase/app/src/main/java/com/example/roomdatabase/MainActivity.kt
package com.example.roomdatabase

import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat

import androidx.core.view.WindowInsetsCompat
import androidx.room.Room
import java.lang.Thread


class MainActivity : AppCompatActivity() {
    private lateinit var database: AppDatabase

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets

        }

        database = Room.databaseBuilder(applicationContext,AppDatabase::class.java,"database-name").build()
        val userDao = database.userDao()

        Thread {
            //thêm một người dùng mới
            val newUser = User(name = "Võ Phi Hùng", email = "acczinsv5@gmail.com", phone = "0352305995")
            userDao.insert(newUser)

            // lấy danh sách tất cả người dùng
            val users = userDao.getAll()

            //sử dụng danh sách người dùng lên textView
            users.forEach{user ->
                Log.d("user", "name: ${user.name}, email: ${user.email}, phone: ${user.phone}")
            }
        }.start()
    }
}